### PR TITLE
StackHCI Registration process Improvements

### PR DESCRIPTION
**Pull request by bniranjanbhat**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/StackHCI/help/Register-AzStackHCI.md [hunk 1]
```diff
@@ -94,7 +95,37 @@ Parameter Sets: (All)
 Aliases:
 
 Required: False
-Position: 9
+Position: 10
```
Comment (isra-fel) [inline src/StackHCI/help/Register-AzStackHCI.md:98]: The order of positional parameters cannot be changed. That was a breaking change. Append new parameters at the end of the parameter list.

Comment (bniranjanbhat) [reply src/StackHCI/help/Register-AzStackHCI.md]: Fixed. thanks !

**Review by VeryEarly — APPROVED** [review]

File: src/StackHCI/ChangeLog.md [hunk 2]
```diff
@@ -18,6 +18,20 @@
...
+* Added support to register hybridConnectivity RP in Canary and Public clouds
+* Added support for Region Specific ServiceEndpoints
+    -Service endpoints are now specified by HCI Cloud services instead of it being hardcoded in the PS module
+* Added support to specify ArcServerResourceGroupName to Register-AzStackHCI cmdlet
+    - Now we can specify the resource group into which the ARC resource gets projected using 'ArcServerResourceGroupName' parameter during registration
+* Check ResourceProvider is already registered before attempting to register during Register-AzStackHCI cmdlet
+    - Register-AzStackHCI cmdlet now does not take any action if required Resource Providers are already registered. This allows users with Roles who do not have permissions to register resource providers to still perform registration operation
+* Supported for writing event log entries from registration cmdlets
+    - Register-AzStackHCI and UnRegister-AzStackHCI flows will now write events to Node log reflecting the status  of registration. Events will be logged under source name "HCI Registration"
+* Updated IMDS Cluster network
```
Comment (bnichms) [inline src/StackHCI/ChangeLog.md:33]: This change was already released

File: src/StackHCI/Az.StackHCI.psm1 [hunk 3]
```diff
@@ -4689,16 +4771,26 @@ function New-Directory{
 #>
 function Invoke-DeploymentModuleDownload{
     # Remote Support
-    New-Variable -Name RemoteSupportPackageUri -Value "https://remotesupportpackages.blob.core.windows.net/packages" -Option Constant -Scope Script
+    New-Variable -Name RemoteSupportPackageUri -Value "https://srsqldatasa.blob.core.windows.net/packages" -Option Constant -Scope Script
```
Comment (bnichms) [inline src/StackHCI/Az.StackHCI.psm1:4774]: Do not bring this change over (or anything related to RS except for logging/etc).

File: src/StackHCI/Az.StackHCI.psm1 [hunk 4]
```diff
@@ -131,33 +126,15 @@ $AzurePPE = "AzurePPE"
...
-
 $Region_EASTUSEUAP = 'eastus2euap'
 
 [hashtable] $ServiceEndpointsAzureCloud = @{
-        $Region_EASTUSEUAP = 'https://eus2euap-azurestackhci-usage.azurewebsites.net';
+        $Region_EASTUSEUAP = 'https://canary.dp.stackhci.azure.com';
         }
 
-$ServiceEndpointAzureCloudFrontDoor = "https://azurestackhci.azurefd.net"
+$ServiceEndpointAzureCloudFrontDoor = "https://dp.stackhci.azure.com"
```
Comment (bnichms) [inline src/StackHCI/Az.StackHCI.psm1:137]: Can you also bring over the Fairfax graph endpoint update?
$GraphServiceApiScopeAzureUSGovernment = "https://graph.windows.net/.default"
should change to
$GraphServiceApiScopeAzureUSGovernment = "https://graph.microsoft.us/.default"

**Pull request merged**